Updated idlejitter-plugin docs (#12012)
netdata/netdata: collectors/idlejitter.plugin/README.md

# idlejitter.plugin

> Idle jitter is the delay of a CPU process.

Reviewer: Not an expert, my understanding of this plugin - it tries to measure CPU preemption latency (task scheduling). @netdata/agent, back me up on this.

Reviewer 2: Both are kind of correct.

The internal measurement is making a call to sleep (really nanosleep or uslep, depending on the system), and then tracking how much longer it takes for the process to be woken up than the time it asked to sleep for. This is functionally ‘lost’ time from the  perspective of the process in question, and high values cause all kinds of problems for realtime and near-realtime applications.

It’s technically a function of preemption latency, and it is also a measure of (one type of) induced delay in processes utilizing such timing functions

I would suggest phrasing it as:

```suggestion
Idle jitter is a measure of delays in timing for user processes caused by scheduling limitations.
```

> ## How Netdata measures idle jitter
> A thread is spawned that requests to sleep for 20000 microseconds (20ms).

Reviewer: ```suggestion
## How Netdata measures idle jitter

A thread is spawned that requests to sleep for 20000 microseconds (20ms).
```

When the system wakes it up, it measures how many microseconds have passed.
The difference between the requested and the actual duration of the sleep, is the idle jitter.
This is done at most 50 times per second, to ensure we have a good average. 

This number is useful:

> - In real-time environments when the CPU jitter can affect the quality of the service (like VoIP media gateways).
> - In cloud infrastructure that can pause the VM or container for a small duration to perform operations at the host.

Reviewer 2: ```suggestion
- In multimedia-streaming environments such as VoIP gateways, where the CPU jitter can affect the quality of the service.
- On time servers and other systems that require very precise timing, where CPU jitter can actively interfere with timing precision.
- On gaming systems, where CPU jitter can cause frame drops and stuttering.
- In cloud infrastructure that can pause the VM or container for a small duration to perform operations at the host.
```

> ## Charts
> idlejitter.plugin generates the idlejitter chart which measures CPU idle jitter in milliseconds lost per second. 

Reviewer: ```suggestion
## Charts

idlejitter.plugin generates the idlejitter chart which measures CPU idle jitter in milliseconds lost per second. 
```

> ## Configuration
> This chart is available without any configuration. 

Reviewer: ```suggestion
## Configuration

This chart is available without any configuration. 
```
